File: packages/android_intent_plus/lib/src/put_classes/put_char.dart
import 'package:android_intent_plus/src/put_classes/base/put_base.dart';

class PutChar extends PutBase<String> {
  PutChar({required String key, required String value})
      : super(key: key, value: value) {
    final length = value.length;
    if (length != 1) {
      throw RangeError.value(
        length,
        "value.length must be 1.",
      );
    }
  }

  @override
  String get javaClass => 'PutChar';

  factory PutChar.fromJson({required String key, required dynamic value}) {
    return PutChar(key: key, value: value as String);
  }

  @override
  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'key': key,
      'javaClass': javaClass,
      'value': value,
    };
  }
}

File: packages/android_intent_plus/android/src/main/java/dev/fluttercommunity/plus/androidintent/Bundle/PutClasses/PutChar.java
package dev.fluttercommunity.plus.androidintent.Bundle.PutClasses;

import android.os.Bundle;
import dev.fluttercommunity.plus.androidintent.Bundle.Constants;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.base.PutBase;
import org.json.JSONException;
import org.json.JSONObject;

public class PutChar extends PutBase<Character> {

  public static final String JAVA_CLASS = "PutChar";

  public PutChar(String key, Character value) {
    super(key, JAVA_CLASS, value);
  }

  public static void convert(Bundle bundle, PutChar putChar) {
    bundle.putChar(putChar.key, putChar.value);
  }

  public static PutChar fromJson(JSONObject jsonObject) throws JSONException {
    return new PutChar(
        jsonObject.getString(Constants.KEY), jsonObject.getString(Constants.VALUE).charAt(0));
  }

  @Override
  public JSONObject toJson() throws JSONException {
    JSONObject jsonObject = new JSONObject();
    jsonObject.put("key", key);
    jsonObject.put("javaClass", JAVA_CLASS);
    jsonObject.put("value", value.toString());
    return jsonObject;
  }
}
